lib/endpointer/cacher.rb

```diff
@@ -10,15 +11,16 @@ def initialize(path)
       initialize_path(path)
     end
 
-    def get(resource)
-      cache_container = retrieve_cache_container(resource)
+    def get(resource, request_body)
+      cache_key = get_cache_key(resource, request_body)
+      cache_container = retrieve_cache_container(cache_key)
       raise Endpointer::Errors::CachedItemNotFoundError unless cache_container.resource == resource
       cache_container.response
```

Comment:
Agree. Needs a bit of a thorough refactoring here. I will handle in a subsequent PR